libr/bin/format/mach0/coresymbolication.c

```diff
@@ -0,0 +1,306 @@
+			ut64 name_off = r_read_le32 (cursor + 0xc);
+			ut64 mangled_name_off = r_read_le32 (cursor + 0x10);
+			sym->unk2 = (st32)r_read_le32 (cursor + 0x14);
+
+			if (cursor + name_off < b || cursor + name_off >= end) {
+				break;
+			}
+			r_str_ncpy (sym->name, (const char *)cursor + name_off, 512);
+
+			if (cursor + mangled_name_off < b || cursor + mangled_name_off >= end) {
+				break;
+			}
+			r_str_ncpy (sym->mangled_name, (const char *)cursor + mangled_name_off, 512);
+
+			cursor += R_CS_EL_SIZE_SYM;
+		}
+	}
+	ut64 start_of_lined_symbols = start_of_symbols + (ut64)hdr->n_symbols * R_CS_EL_SIZE_SYM;
+	if (hdr->n_lined_symbols) {
+		result->lined_symbols = R_NEWS0 (RCoreSymCacheElementLinedSymbol, hdr->n_lined_symbols);
+		if (!result->lined_symbols) {
+			goto beach;
+		}
+		size_t i;
+		ut8 *cursor = b + start_of_lined_symbols;
+		for (i = 0; i < hdr->n_lined_symbols && cursor + R_CS_EL_SIZE_LSYM <= end; i++) {
+			RCoreSymCacheElementLinedSymbol *lsym = &result->lined_symbols[i];
+			lsym->sym.paddr = r_read_le32 (cursor);
+			lsym->sym.size = r_read_le32 (cursor + 0x4);
+			lsym->sym.unk1 = r_read_le32 (cursor + 0x8);
+			ut64 name_off = r_read_le32 (cursor + 0xc);
+			ut64 mangled_name_off = r_read_le32 (cursor + 0x10);
+			lsym->sym.unk2 = (st32)r_read_le32 (cursor + 0x14);
+			ut64 file_name_off = r_read_le32 (cursor + 0x18);
+			lsym->flc.line = r_read_le32 (cursor + 0x1c);
+			lsym->flc.col = r_read_le32 (cursor + 0x20);
+
+			if (cursor + name_off < b || cursor + name_off >= end) {
+				break;
+			}
+			r_str_ncpy (lsym->sym.name, (const char *)cursor + name_off, 512);
+
+			if (cursor + mangled_name_off < b || cursor + mangled_name_off >= end) {
+				break;
+			}
+			r_str_ncpy (lsym->sym.mangled_name, (const char *)cursor + mangled_name_off, 512);
+
+			if (cursor + file_name_off < b || cursor + file_name_off >= end) {
+				break;
+			}
+			r_str_ncpy (lsym->flc.file, (const char *)cursor + file_name_off, 512);
+
+			cursor += R_CS_EL_SIZE_LSYM;
+
+			meta_add_fileline (bf, r_coresym_cache_element_pa2va (result, lsym->sym.paddr), lsym->sym.size, &lsym->flc);
+		}
+	}
+	ut64 start_of_line_info = start_of_lined_symbols + (ut64)hdr->n_lined_symbols * R_CS_EL_SIZE_LSYM;
+	if (hdr->n_line_info) {
+		result->line_info = R_NEWS0 (RCoreSymCacheElementLineInfo, hdr->n_line_info);
+		if (!result->line_info) {
+			goto beach;
+		}
+		size_t i;
+		ut8 *cursor = b + start_of_line_info;
+		for (i = 0; i < hdr->n_line_info && cursor + R_CS_EL_SIZE_LINFO <= end; i++) {
+			RCoreSymCacheElementLineInfo *info = &result->line_info[i];
+			info->paddr = r_read_le32 (cursor);
+			info->size = r_read_le32 (cursor + 4);
+			ut64 file_name_off = r_read_le32 (cursor + 8);
+			info->flc.line = r_read_le32 (cursor + 0xc);
+			info->flc.col = r_read_le32 (cursor + 0x10);
+
+			if (cursor + file_name_off < b || cursor + file_name_off >= end) {
+				break;
+			}
+			r_str_ncpy (info->flc.file, (const char *)cursor + file_name_off, 512);
```

Comment:
dont do 512, use sizeof() but as mentioned below i think this is a waster of memory when lots of symbols are there, so it will be better to use a string pool. or for now just use the heap